bottlecap/src/metrics/flusher.rs

```diff
@@ -16,31 +16,30 @@ impl Flusher {
     }
 
     pub async fn flush(&mut self) {
-        let locked_aggr = &mut self.aggregator.lock().expect("lock poisoned");
-        let current_points = locked_aggr.to_series();
-        let current_distribution_points = locked_aggr.distributions_to_protobuf();
-        if !current_points.series.is_empty() {
-            debug!("flushing {} series to datadog", current_points.series.len());
-            match &self.dd_api.ship_series(&current_points).await {
+        let all_series;
+        let all_distributions;
+        {
+            let locked_aggr = &mut self.aggregator.lock().expect("lock poisoned");
+            all_series = locked_aggr.consume_metrics();
+            all_distributions = locked_aggr.consume_distributions();
+        }
```

Comment:
Seems like this should just be:

```suggestion
        let (all_series, all_distributions) = {
            let mut aggregator = self.aggregator.lock().expect("lock poisoned");
            (aggregator.consume_metrics(), aggregator.consume_distributions())
        };
```

Reply:
oh thanks, the mut was in the wrong spot, ty!